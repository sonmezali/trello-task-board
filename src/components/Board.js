import React from 'react'
import Card from "./Card"

const Board=(props)=> {
  
    const handleDrop=(e)=>{
        e.preventDefault();
        const cardId = e.dataTransfer.getData('card_id');       
        props.handleDrop(cardId, props.id)         
        }

    const handleDragOver=(e)=>{
        e.preventDefault(); 
        }

    const handleAddCard=()=>{
        const newCard= { id:`c${props.cards.length+1}`, title: `Task ${props.cards.length+1}`, boardId: `${props.id}` }
        props.setCards([...props.cards,newCard]);
        localStorage.setItem('cards', JSON.stringify([...props.cards,newCard]))
        }

    return (
        <div id={props.id}      
            className={props.className}
            onDrop={handleDrop}
            onDragOver={handleDragOver}
        >
        <div className="board-title">{props.title}</div>
            {props.filteredCards.map(card => <Card key={card.id} title={card.title}  id={card.id}
            cards={props.cards} setCards={props.setCards}  
            className="card"/>)}
            <div className="add-card-btn"  onClick={handleAddCard}><span>+Add Card</span></div>
        </div>
    )
}

export default Board
