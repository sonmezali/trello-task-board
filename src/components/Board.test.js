import React from 'react';
import ReactDom,{unmountComponentAtNode} from 'react-dom';
import Board from './Board'
import {render} from '@testing-library/react';
import renderer from 'react-test-renderer';




describe('test Board component', () => {

  let container = null;
  beforeEach(() => {  
    container = document.createElement("div");
  });

  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  test("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDom.render(<Board></Board>, div);
  });

  test('matches snapshot', () => {
    const tree = renderer.create(<Board></Board>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders button correctly', () => {
    const {getByTestId} = render(<Board></Board>)
    expect(getByTestId('add-card')).toHaveTextContent('+Add Card')    
  })

})

