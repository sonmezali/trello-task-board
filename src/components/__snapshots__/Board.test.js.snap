// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Board component matches snapshot 1`] = `
<div
  onDragOver={[Function]}
  onDrop={[Function]}
>
  <div
    className="board-title"
  />
  <div
    className="add-card-btn"
    data-testid="add-card"
    onClick={[Function]}
  >
    <span>
      +Add Card
    </span>
  </div>
</div>
`;
